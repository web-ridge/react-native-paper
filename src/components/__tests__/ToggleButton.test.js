import * as React from 'react';
import renderer from 'react-test-renderer';
import ToggleButton from '../ToggleButton/ToggleButton';

it('renders toggle button', () => {
  const tree = renderer
    .create(<ToggleButton status="checked" onPress={() => {}} icon="heart" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders disabled toggle button', () => {
  const tree = renderer
    .create(
      <ToggleButton
        disabled
        value="toggle"
        status="checked"
        onValueChange={() => {}}
        icon="heart"
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders unchecked toggle button', () => {
  const tree = renderer
    .create(
      <ToggleButton
        disabled
        status="unchecked"
        onValueChange={() => {}}
        icon="heart"
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
