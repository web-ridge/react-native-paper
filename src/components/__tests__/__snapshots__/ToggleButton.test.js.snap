// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disabled toggle button 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
      "selected": true,
    }
  }
  accessible={true}
  centered={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 6,
      "left": 6,
      "right": 6,
      "top": 6,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      false,
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 6,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 18,
          "height": 36,
          "width": 36,
        },
        Object {
          "opacity": 0.32,
        },
        Array [
          Object {
            "height": 42,
            "margin": 0,
            "width": 42,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, .08)",
            "borderColor": "rgba(0, 0, 0, 0.29)",
            "borderRadius": 4,
          },
          undefined,
        ],
      ],
    ]
  }
>
  <View>
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders toggle button 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": undefined,
      "selected": true,
    }
  }
  accessible={true}
  centered={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 6,
      "left": 6,
      "right": 6,
      "top": 6,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      false,
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 6,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 18,
          "height": 36,
          "width": 36,
        },
        undefined,
        Array [
          Object {
            "height": 42,
            "margin": 0,
            "width": 42,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, .08)",
            "borderColor": "rgba(0, 0, 0, 0.29)",
            "borderRadius": 4,
          },
          undefined,
        ],
      ],
    ]
  }
>
  <View>
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders unchecked toggle button 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": true,
      "selected": false,
    }
  }
  accessible={true}
  centered={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 6,
      "left": 6,
      "right": 6,
      "top": 6,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "position": "relative",
      },
      false,
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 6,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 18,
          "height": 36,
          "width": 36,
        },
        Object {
          "opacity": 0.32,
        },
        Array [
          Object {
            "height": 42,
            "margin": 0,
            "width": 42,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "rgba(0, 0, 0, 0.29)",
            "borderRadius": 4,
          },
          undefined,
        ],
      ],
    ]
  }
>
  <View>
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 24,
          },
          Array [
            Object {
              "lineHeight": 24,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
